packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts

```diff
@@ -193,7 +194,7 @@ export class SnowflakeWarehouseClient extends WarehouseBaseClient<CreateSnowflak
         let connection: Connection;
         try {
             connection = createConnection(this.connectionOptions);
-            await Util.promisify(connection.connect)();
+            await Util.promisify(connection.connect.bind(connection))();
```

Comment:
this should fix the `this...` is undefined, thing, right ?

Reply:
yes!

Reply:
check this discussion for the context: https://github.com/snowflakedb/snowflake-connector-nodejs/issues/929#issuecomment-2393283248